test/sandbox/git.go

```diff
@@ -23,24 +23,44 @@ import (
 	"github.com/mineiros-io/terramate/test"
 )
 
-const defRemote = "origin"
-const defRemoteBranch = "main"
-const defLocalBranch = "main"
+// GitConfig configures the sandbox's git repository.
+type GitConfig struct {
+	LocalBranchName         string
+	DefaultRemoteName       string
+	DefaultRemoteBranchName string
+
+	repoDir string
+}
 
 // Git is a git wrapper that makes testing easy by handling
 // errors automatically, failing the caller test.
 type Git struct {
 	t          *testing.T
 	g          *git.Git
+	cfg        GitConfig
 	localrepo  string
 	remoterepo string
 }
 
-func NewGit(t *testing.T, basedir string) Git {
-	return Git{
+// NewGit creates a new git wrapper using sandbox defaults.
+func NewGit(t *testing.T, repodir string) *Git {
+	cfg := defaultGitConfig()
+	cfg.repoDir = repodir
+
+	return &Git{
+		t:         t,
+		cfg:       cfg,
+		g:         test.NewGitWrapper(t, repodir, []string{}),
+		localrepo: repodir,
```

Comment:
Isn't localrepo now redundant with cfg.repoDir ? maybe we could always just reference cfg.repoDir ?